tests/bgp/test_route_aggregation.py

```diff
@@ -0,0 +1,187 @@
+
+    # Configure summary-only route aggregation
+    agg_configuration(True, gather_info['dut_asn'], gather_info['duthost'],
+                      gather_info['cli_options'], "summary-only", "summary-only")
+    logger.debug(gather_info['duthost'].shell('vtysh -c "show run bgp"', module_ignore_errors=True)['stdout'])
+    verify_route_agg(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], "1", False)
+
+    # Remove the aggregate configuration
+    agg_configuration(False, gather_info['dut_asn'], gather_info['duthost'],
+                      gather_info['cli_options'], "summary-only", "summary-only")
+    verify_route_agg(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], "5", False)
+    check_baseline(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], num_v4_routes,
+                   num_v6_routes)
+
+    # Configure as-set summary-only route aggregation
+    agg_configuration(True, gather_info['dut_asn'], gather_info['duthost'],
+                      gather_info['cli_options'], "as-set summary-only", "as-set summary-only")
+    logger.debug(gather_info['duthost'].shell('vtysh -c "show run bgp"', module_ignore_errors=True)['stdout'])
+    verify_route_agg(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], "1", False,
+                     as_set=True)
+
+    # Remove the aggregate configuration
+    agg_configuration(False, gather_info['dut_asn'], gather_info['duthost'],
+                      gather_info['cli_options'], "as-set summary-only", "as-set summary-only")
+    verify_route_agg(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], "5", False)
+    check_baseline(gather_info['duthost'], gather_info['neigh_ip_v4'], gather_info['neigh_ip_v6'], num_v4_routes,
+                   num_v6_routes)
+
+    # Configure route aggregation with suppress-map
+    # Create prefix lists to be used
+    cmd = 'vtysh{} -c "config" -c "ip prefix-list SUPPRESS_V4 permit 1.1.1.1/24" \
+            -c "ip prefix-list SUPPRESS_V4 permit 1.1.2.1/24" \
+            -c "ipv6 prefix-list SUPPRESS_V6 permit 1:1:1:1::/64" \
+            -c "ipv6 prefix-list SUPPRESS_V6 permit 1:1:1:2::/64"' \
+            .format(gather_info['cli_options'])
+    logger.debug(gather_info['duthost'].shell(cmd, module_ignore_errors=True))
+
+    # Create route maps
+    cmd = 'vtysh{} -c "config" -c "route-map SUPPRESS_RM_V4 permit 10" -c "match ip address prefix-list SUPPRESS_V4"' \
+          .format(gather_info['cli_options'])
+    gather_info['duthost'].shell(cmd, module_ignore_errors=True)
+    cmd = 'vtysh{} -c "config" -c "route-map SUPPRESS_RM_V6 permit 10" \
+           -c "match ipv6 address prefix-list SUPPRESS_V6"' \
+          .format(gather_info['cli_options'])
+    gather_info['duthost'].shell(cmd, module_ignore_errors=True)
+    agg_configuration(True, gather_info['dut_asn'], gather_info['duthost'],
```

Comment:
same here.

Reply:
The creation of the route maps is done twice with very different commands and setups for the maps.  I don't see a way to combine into a function.